superset-frontend/src/views/CRUD/welcome/EmptyState.tsx

```diff
@@ -0,0 +1,144 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import Button from 'src/components/Button';
+import { Empty } from 'src/common/components';
+import { t, styled } from '@superset-ui/core';
+import Icon from 'src/components/Icon';
+import { IconContainer } from '../utils';
+
+interface EmptyStateProps {
+  tableName: string;
+  tab?: string;
+}
+
+const ButtonContainer = styled.div`
+  Button {
+    svg {
+      color: ${({ theme }) => theme.colors.grayscale.light5};
+    }
+  }
+`;
+
+export default function EmptyState({ tableName, tab }: EmptyStateProps) {
+  const mineRedirects = {
+    DASHBOARDS: '/dashboard/new',
+    CHARTS: '/chart/add',
+    SAVED_QUERIES: '/superset/sqllab',
+  };
+  const favRedirects = {
+    DASHBOARDS: '/dashboard/list/',
+    CHARTS: '/chart/list',
+    SAVED_QUERIES: '/savedqueryview/list/',
+  };
+  const tableIcon = {
+    RECENTS: 'union.png',
+    DASHBOARDS: 'empty-dashboard.png',
+    CHARTS: 'empty-charts.png',
+    SAVED_QUERIES: 'empty-queries.png',
+  };
+  const mine = (
+    <div>{`No ${
+      tableName === 'SAVED_QUERIES'
+        ? t('saved queries')
+        : t(`${tableName.toLowerCase()}`)
+    } yet`}</div>
+  );
+  const recent = (
+    <div className="no-recents">
+      {(() => {
+        if (tab === 'Viewed') {
+          return t(
+            `Recently viewed charts, dashboards, and saved queries will appear here`,
+          );
+        }
+        if (tab === 'Created') {
+          return t(
+            'Recently created charts, dashboards, and saved queries will appear here',
+          );
+        }
+        if (tab === 'Examples') {
+          return t(
+            `Recent example charts, dashboards, and saved queries will appear here`,
+          );
+        }
+        if (tab === 'Edited') {
+          return t(
+            `Recently edited charts, dashboards, and saved queries will appear here`,
+          );
+        }
+        return null;
+      })()}
+    </div>
+  );
+  // Mine and Recent Activity(all tabs) tab empty state
+  if (tab === 'Mine' || tableName === 'RECENTS') {
```

Comment:
Nit: can we make the tab name all lowercase to have some consistency?

```
if (tab === 'mine' || tableName === 'recents') {
```

The idea is to not have to think about which case to use.